File: CommonBase/src/main/java/com/common/base/mvvm/MvvmActivity.java
package com.common.base.mvvm;

import android.os.Bundle;

import com.common.base.BaseActivity;
import com.common.base.ability.IViewModel;

import androidx.lifecycle.ViewModel;

/**
 * MVVM模式Activity
 *
 * @author LiuFeng
 * @data 2021/9/22 16:20
 */
public abstract class MvvmActivity<V extends ViewModel> extends BaseActivity implements IViewModel<V> {

    protected V viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        this.viewModel = this.createViewModel();
        super.onCreate(savedInstanceState);
    }
}

File: CommonBase/src/main/java/com/common/base/mvvm/MvvmActivity.kt
package com.common.base.mvvm

import androidx.lifecycle.ViewModel
import com.common.base.BaseActivity

/**
 * MVVM模式Activity
 *
 * @author LiuFeng
 * @data 2021/9/22 16:20
 */
abstract class MvvmActivity<VM : ViewModel> : BaseActivity() {
    protected abstract val viewModel: VM
}